edge-modules/load-gen/Program.cs

```diff
@@ -91,8 +93,28 @@ static async Task SendEventAsync(ModuleClient client, Guid batchId, string track
                 message.Properties.Add("batchId", batchId.ToString());
                 message.Properties.Add("trackingId", trackingId);
 
+                // send a copy of message to the Test Result Coordinator if applicable
+                if(Settings.Current.TestResultCoordinatorUrl != null)
+                {
+                    Uri testResultCoordinatorUrl = Settings.Current.TestResultCoordinatorUrl;
+                    TestResultCoordinatorClient trcClient = new TestResultCoordinatorClient { BaseUrl = testResultCoordinatorUrl.AbsoluteUri };
+                    await ReportStatus(trcClient, Settings.Current.TrackingId, message.ToString());
+                }
+
                 await client.SendEventAsync(Settings.Current.OutputName, message);
             }
         }
+
+        static async Task ReportStatus(TestResultCoordinatorClient trcClient, string source, string message)
+        {
+            try
+            {
+                await trcClient.ReportResultAsync(new TestOperationResult { Source = source, Result = message, CreatedAt = DateTime.UtcNow, Type = "LoadGen" });
```

Comment:
Wrong value for Type and Result.

Reply:
Please create a helper method in Microsoft.Azure.Devices.Edge.ModuleUtil.csproj to construct the result value e.g. ConstructSimpleResult(string trackingId, string batchId, string sequenceNumber); therefore other modules can reuse.

Reply:
you should be able to get trackingId, batchId, sequenceNumber from message.Properties.